src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -212,10 +212,13 @@ void RadioSpinel<InterfaceType, ProcessContextType>::Init(bool aResetRadio, bool
     if (aResetRadio)
     {
         SuccessOrExit(error = SendReset());
+        SuccessOrExit(error = WaitResponse());
+        VerifyOrExit(mIsReady, error = OT_ERROR_FAILED);
+    }
+    else
```

Comment:
When `aResetRadio` is false, the host assumes the RCP is reset through other approaches(e.g. reset PIN) instead of spinel, so the reset event is always expected to keep host and RCP in sync. The `no-reset` option is used to avoid double resets.

Reply:
@latheepanm - any thoughts on @bukepo's comment?

Reply:
@romacdon I actually put my thoughts in #5712 and maybe I should have put it here as well. 

@bukepo For our application, it would be desirable to not send a reset to the RCP when an app launches. Is there any other way to ensure that they are in sync? Is there a recommended set/order of commands that should be sent to the RCP, that is not part of the initialization?